### PR TITLE
feat: Support JSON-LD for Verifiable Credentials

### LINKED ISSUE #952: Support JSON-LD for Verifiable Credentials
https://www.w3.org/TR/vc-data-model/#json-ld

### PR DESCRIPTION
**Pull request by kdimak**
closes #952, #978

Signed-off-by: Dima <dkinoshenko@gmail.com>

By default, JSON-LD validation is made when there is more than one `@context` of VC is defined.

### CODE REVIEW

File: pkg/doc/verifiable/credential.go [hunk 1]
```diff
@@ -587,56 +608,50 @@ func NewCredential(vcData []byte, opts ...CredentialOpt) (*Credential, []byte, e
...
 }
 
-func postValidateCredential(vc *Credential, vcOpts *credentialOpts) error {
+func validateCredential(vc *Credential, vcBytes []byte, vcOpts *credentialOpts) error {
 	// Credential and type constraint.
 	switch vcOpts.modelValidationMode {
-	case jsonldValidation:
-		// todo support JSON-LD validation (https://github.com/hyperledger/aries-framework-go/issues/952)
+	case adaptedValidation:
+		err := vc.validateJSONSchema(vcBytes, vcOpts)
```
Comment (troyronda) [inline pkg/doc/verifiable/credential.go]: Why do we also need to validateJSONSchema when len(vc.Context) > 1?

File: go.mod [hunk 2]
```diff
@@ -27,4 +27,7 @@ require (
 	nhooyr.io/websocket v1.7.4
 )
 
+// Temporary workaround to support protected (json-gold issue 22)
+replace github.com/piprate/json-gold v0.2.0 => github.com/trustbloc/json-gold v0.2.1-0.20191211165818-ef1958eae4be
```
Comment (troyronda) [inline go.mod]: I assume this branch should work: https://github.com/piprate/json-gold/tree/json11-dec-2019

Comment (kdimak) [reply go.mod]: Agree, I will change this!

File: pkg/doc/verifiable/credential.go (hunk 1 above)
Comment (kdimak) [reply pkg/doc/verifiable/credential.go]: Agree! I added a comment here and also renamed the validation mode to `combinedValidation`. Also, a more detailed explanation of `combinedValidation` mode is provided at the definition.

File: pkg/doc/verifiable/credential.go [hunk 3]
```diff
@@ -464,10 +478,12 @@ func WithJSONLDValidation() CredentialOpt {
 }
 
 // WithBaseContextValidation validates that only the fields and values (when applicable) are present
-// in the document. No extra fields are allowed (outside of credentialSubject).
+// in the document. No extra fields are allowed (outside of credentialSubject). Strict JSON Schema validation is made
+// (i.e. fields not defined in the base JSON Schema are not allowed).
 func WithBaseContextValidation() CredentialOpt {
 	return func(opts *credentialOpts) {
 		opts.modelValidationMode = baseContextValidation
+		opts.strictValidation = true
```
Comment (troyronda) [inline pkg/doc/verifiable/credential.go]: i am not sure that this line works without flexibility in the credentialsubject.

Comment (kdimak) [reply pkg/doc/verifiable/credential.go]: Corrected.

**Pull request merged**